napari/settings/_base.py

```diff
@@ -364,8 +367,32 @@ def config_file_settings_source(
 
     # if the config has a `sources` list, read those too and merge.
     sources = list(getattr(settings.__config__, 'sources', []))
-    if config_path:
+    if config_path and Path(config_path).exists():
         sources.append(config_path)
+    elif config_path and Path(config_path).parent.parent.exists():
+        # check for previous version directory
+        napari_dir = Path(config_path).parent.parent
+        napari_versions = (
+            (version.Version(dir.name), dir)
+            for dir in napari_dir.iterdir()
+            if dir.is_dir()
+            and isinstance(version.parse(dir.name), version.Version)
+        )
+        napari_version = version.parse(napari.utils._appdirs.version_string)
+        napari_lower_version = sorted(
+            ((v, d) for v, d in napari_versions if v < napari_version),
+            reverse=True,
+        )
+        if napari_lower_version:
```

Comment:
Optional: I found this code block a little hard to read the first time through. This function is also already pretty large. I'd consider writing a small helper function that clearly states what you care about (i.e. the subdirectory that corresponds to the previous version).

```python
def _previous_version_subdirectory(root: Path) -> Optional[Path]:
    ...
```

Reply:
Where would I put this function?

Reply:
My personal approach is to put it at the bottom of the same module as a private function (i.e. with a leading underscore). Looks like it already has some friends in this module (e.g. `_remove_bad_keys`), so it won't be lonely :)